tests/drivers/gpio/gpio_basic_api/boards/frdm_mcxe31b.overlay

```diff
@@ -0,0 +1,29 @@
+/*
```

Comment:
why add frdm_mcxe31b in this pr? besides, there is no need to add separated overlay for gpio if you have arduino_header defined, D2D3 is default enabled